src/GiniImpurity.jl

```diff
@@ -77,12 +77,16 @@ Args:
 Returns:
     A tuple containing the index of the best feature and the best split value.
 """
-function find_best_split(data::Matrix{T}, labels::Vector{L}) where {T, L}  
+function find_best_split(data::Matrix{T}, labels::Vector{L}, num_features_to_use::Int=-1) where {T, L}  
     best_gini = Inf  
     best_feature_index = -1  
     best_feature_value = -1  
     num_features = size(data, 2)
-    for feature_index in 1:num_features  
+    features_to_use = 1:num_features
+    if (num_features_to_use != -1)
+        features_to_use = sample(1:num_features, num_features_to_use, replace=true)
```

Comment:
shouldn't the choice of features here be without replacement?

Reply:
I am unsure. Is there some universally valid implementation of the algorithm?

Reply:
I think if we select features with replacements we might select the same feature more than once
So if num_features_to_use is 3 we actually might use the information from only 1/2 features when finding the best split. Some examples I saw online used it without replacements as well
For example: https://youtu.be/NxEHSAfFlK8?si=jtKwL2SZRg8DQUQo&t=996

Reply:
Then i will change it